packages/interactivity-router/src/index.ts

```diff
@@ -108,20 +129,51 @@ const regionsToVdom: RegionsToVdom = async ( dom, { vdom, url } = {} ) => {
 		Promise.all( preloadScriptModules( dom ) ),
 	] );
 
-	return { regions, styles, scriptModules, title, initialData, url };
+	return {
+		regions,
+		regionsToAttach,
+		styles,
+		scriptModules,
+		title,
+		initialData,
+		url,
+	};
 };
 
 // Render all interactive regions contained in the given page.
 const renderRegions = ( page: Page ) => {
 	applyStyles( page.styles );
 
+	// Clone regionsToAttach.
+	const regionsToAttach = { ...page.regionsToAttach };
+
 	batch( () => {
 		populateServerData( page.initialData );
 		document.querySelectorAll( regionsSelector ).forEach( ( region ) => {
-			const id = region.getAttribute( regionAttr );
+			const { id } = parseRegionAttribute( region );
 			const fragment = getRegionRootFragment( region );
 			render( page.regions[ id ], fragment );
+			// If this is an attached region, remove it from the list.
+			delete regionsToAttach[ id ];
 		} );
+
+		// Render unattached regions.
+		for ( const id in regionsToAttach ) {
+			const parent = document.querySelector( regionsToAttach[ id ] );
+
+			// Get the type from the vnode. If wrapped with Directives, get the
+			// original type from `props.type`.
+			const { props, type } = page.regions[ id ];
+			const elementType = typeof type === 'function' ? props.type : type;
+
+			// Create an element with the obtained type where the region will be
+			// rendered. The type should match the one of the root vnode.
+			const region = document.createElement( elementType );
+			parent.appendChild( region );
```

Comment:
The types of the element created here and the vnode should match to prevent Preact errors.